File: backend/app/schemas/risk.py
from __future__ import annotations

from pydantic import BaseModel, Field


class RiskSummary(BaseModel):
    min: float = 0.0
    max: float = 1.0
    mean: float = 0.0


class RiskAlert(BaseModel):
    level: str = Field(..., description="LOW, MODERATE, HIGH, VERY_HIGH")
    threshold: float = 0.7


class RiskBounds(BaseModel):
    west: float
    south: float
    east: float
    north: float


class RiskResponse(BaseModel):
    predictionId: str
    regionId: str = "bihar-nepal"
    hazard: str = "flood"
    risk: RiskSummary
    resolutionMeters: float
    crs: str
    bounds: RiskBounds | None = None
    satellite: str = "Sentinel-1"
    sourcePassTimestamp: str
    predictionTimestamp: str
    processingLagSeconds: int
    predictionHorizonHours: int
    modelId: str
    modelVersion: str
    alert: RiskAlert
    rasterUrl: str | None = None
    previewUrl: str | None = None
    gridUrl: str | None = None
    observationWindowScenes: list[str] | None = None
    gridShape: list[int] | None = None


class RiskHistoryPoint(BaseModel):
    predictionId: str
    predictionTimestamp: str
    mean: float
    alertLevel: str


class RiskGridCell(BaseModel):
    lat: float
    lon: float
    value: float


class RiskGridResponse(BaseModel):
    rows: int
    cols: int
    cells: list[RiskGridCell]

File: frontend/src/types/risk.ts
import type { RegionBounds } from './region';

export interface RiskSummary {
  min: number;
  max: number;
  mean: number;
}

export interface RiskAlert {
  level: 'LOW' | 'MODERATE' | 'HIGH' | 'VERY_HIGH' | string;
  threshold: number;
}

export interface RiskSnapshot {
  predictionId: string;
  regionId: string;
  hazard: string;
  risk: RiskSummary;
  resolutionMeters: number;
  crs: string;
  bounds: RegionBounds | null;
  satellite: string;
  sourcePassTimestamp: string;
  predictionTimestamp: string;
  processingLagSeconds: number;
  predictionHorizonHours: number;
  modelId: string;
  modelVersion: string;
  alert: RiskAlert;
  rasterUrl: string | null;
  previewUrl: string | null;
  gridUrl: string | null;
  observationWindowScenes: string[] | null;
  gridShape: [number, number] | null;
  note: string | null;
}

export interface RiskHistoryPoint {
  predictionId: string;
  predictionTimestamp: string;
  mean: number;
  alertLevel: string;
}

export interface RiskGridCell {
  lat: number;
  lon: number;
  value: number;
}

export interface RiskGrid {
  rows: number;
  cols: number;
  cells: RiskGridCell[];
}
